gcc/rust/rust-session-manager.cc

```diff
@@ -547,6 +549,22 @@ Session::parse_file (const char *filename)
       dump_type_resolution (hir);
     }
 
+  // liveness analysis
+  std::set<HirId> live_symbols = Analysis::Liveness::Analysis (hir);
+  if (options.dump_option_enabled (CompileOptions::ANALYSIS_DUMP))
+    {
+      // I have no idea about what should be dumped. Maybe some bit-vectors?
```

Comment:
This is an interesting one thanks for thinking about it. We also have an incomplete name resolution dump. What if we simply emit the IDs for the live/dead code sets. HIR dumps when complete should/must contain the NodeMappings information which can be used for reference.

Reply:
Dumps should be a separate PR/commit i think

Reply:
Yeah, this is a sparse analysis currently, dump hirid for live/dead code sets might be a suitable way. I will separate this dump part for other PR.